specification/compute/resource-manager/Microsoft.Compute/stable/2020-05-01/disk.json

```diff
@@ -0,0 +1,2456 @@
+        },
+        "previousKeys": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/KeyVaultAndKeyReference"
+          },
+          "description": "A readonly collection of key vault keys previously used by this disk encryption set while a key rotation is in progress. It will be empty if there is no ongoing key rotation."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The disk encryption set provisioning state."
+        }
+      }
+    },
+    "EncryptionSettingsCollection": {
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Set this flag to true and provide DiskEncryptionKey and optional KeyEncryptionKey to enable encryption. Set this flag to false and remove DiskEncryptionKey and KeyEncryptionKey to disable encryption. If EncryptionSettings is null in the request object, the existing settings remain unchanged."
+        },
+        "encryptionSettings": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EncryptionSettingsElement"
+          },
+          "description": "A collection of encryption settings, one for each disk volume."
+        },
+        "encryptionSettingsVersion": {
+          "type": "string",
+          "description": "Describes what type of encryption is used for the disks. Once this field is set, it cannot be overwritten. '1.0' corresponds to Azure Disk Encryption with AAD app.'1.1' corresponds to Azure Disk Encryption."
+        }
+      },
+      "required": [
+        "enabled"
+      ],
+      "description": "Encryption settings for disk or snapshot"
+    },
+    "EncryptionSettingsElement": {
+      "properties": {
+        "diskEncryptionKey": {
+          "$ref": "#/definitions/KeyVaultAndSecretReference",
+          "description": "Key Vault Secret Url and vault id of the disk encryption key"
+        },
+        "keyEncryptionKey": {
+          "$ref": "#/definitions/KeyVaultAndKeyReference",
+          "description": "Key Vault Key Url and vault id of the key encryption key. KeyEncryptionKey is optional and when provided is used to unwrap the disk encryption key."
+        }
+      },
+      "description": "Encryption settings for one disk volume."
+    },
+    "KeyVaultAndSecretReference": {
+      "properties": {
+        "sourceVault": {
+          "$ref": "#/definitions/SourceVault",
+          "description": "Resource id of the KeyVault containing the key or secret"
+        },
+        "secretUrl": {
+          "type": "string",
+          "description": "Url pointing to a key or secret in KeyVault"
+        }
+      },
+      "required": [
+        "secretUrl",
+        "sourceVault"
+      ],
+      "description": "Key Vault Secret Url and vault id of the encryption key "
+    },
+    "KeyVaultAndKeyReference": {
+      "properties": {
+        "sourceVault": {
```

Comment:
Can you make this an optional property.
This Ref is shared between ADE and SSE CMK. And the property is required by ADE but optional for CMK now.
I am not sure if we need to define another Ref for CMK with this removed or just making it optional is sufficient

Reply:
only keyUrl is required.. required property  below shows the fields compulsory